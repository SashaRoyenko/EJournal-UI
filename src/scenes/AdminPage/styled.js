import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 15px;
`;

export const Main = styled.div`
  display: flex;
`;
