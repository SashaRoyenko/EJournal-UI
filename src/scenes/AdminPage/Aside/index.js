import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper, MenuItem } from './styled';
import getMenuItems from './menuItems';

export default class Aside extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      menuItems: getMenuItems(props.matchPath),
    };
  }

  render() {
    const { menuItems } = this.state;
    return (
      <Wrapper>
        {
          menuItems.map(menuItem => (
            <MenuItem key={menuItem.id} to={menuItem.to}>{menuItem.title}</MenuItem>
          ))
        }
      </Wrapper>
    );
  }
}

Aside.propTypes = {
  matchPath: PropTypes.string,
};

Aside.defaultProps = {
  matchPath: '/',
};
