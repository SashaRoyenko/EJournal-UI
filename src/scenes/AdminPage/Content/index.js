import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
import getRoutes from './contentTabs';
import { Wrapper, Header } from './styled';

export default class Content extends React.Component {
  constructor(props) {
    super(props);

    const {matchPath} = props;

    this.state = {
      routes: getRoutes(matchPath),
    };
  }

  renderHeader(){
    const { routes } = this.state;
    return (
      routes.map(route => (
        <Route key={route.id} path={route.path} component={() => <Header>{route.title}</Header>}/>
      ))
    )
  }

  renderContent(){
    const { routes } = this.state;
    return (
      routes.map(route => (
        <Route key={route.id} path={route.path} component={route.component}/>
      ))
    )
  }

  render() {
    return (
      <Wrapper>
        {this.renderHeader()}
        {this.renderContent()}
      </Wrapper>
    );
  }
}

Content.propTypes = {
  matchPath: PropTypes.string,
};

Content.defaultProps = {
  matchPath: '/',
};

