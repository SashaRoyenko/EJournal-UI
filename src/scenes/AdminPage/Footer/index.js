import React from 'react';
import { Wrapper, Label } from './styled';

export default class Footer extends React.Component {
  render() {
    return (
      <Wrapper>
        <Label>Â© E-JOURNAL. ROBOSH 2020</Label>
      </Wrapper>
    );
  }
}
