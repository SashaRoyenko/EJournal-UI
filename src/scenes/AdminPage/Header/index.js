import React from 'react';
import { Wrapper, HomeLink, LoginWrapper, Label, LogOutButton } from './styled';

export default class Header extends React.Component {
  render() {
    return (
      <Wrapper>
        <HomeLink to='/'>E-JOURNAL</HomeLink>
        <LoginWrapper>
          <Label>
            Ви увійшли як: <Label highlighted>SuperAdmin</Label>
          </Label>
          <LogOutButton>Вийти</LogOutButton>
        </LoginWrapper>
      </Wrapper>
    );
  }
}
