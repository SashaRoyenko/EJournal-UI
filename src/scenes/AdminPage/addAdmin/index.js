import React from 'react';
import {Button, Input, InputItem, ItemTitle, Wrapper} from './styled';

export default class AddAdmin extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            schools: ["1", "2", "3", "Na"],
        };
    }

    render() {
        const {schools} = this.state;
        return (
            <form>
                <Wrapper>
                    <InputItem>
                        <ItemTitle>Прізвище</ItemTitle>
                        <Input type="text" placeholder="   Прізвище"/>
                    </InputItem>
                    <InputItem>
                        <ItemTitle>Ім&apos;я</ItemTitle>
                        <Input type="text" placeholder="   Ім'я"/>
                    </InputItem>
                    <InputItem>
                        <ItemTitle>Навчальний заклад</ItemTitle>
                        <select name="school">
                            {schools.map((school) => <option key={"#"} value="#">{school}</option>)}
                        </select>
                    </InputItem>
                    <InputItem>
                        <ItemTitle>Електронна пошта</ItemTitle>
                        <Input type="text" placeholder="   E-mail"/>
                    </InputItem>
                    <InputItem>
                        <ItemTitle>Пароль</ItemTitle>
                        <Input type="password" placeholder="   ******"/>
                    </InputItem>
                </Wrapper>
                <Button>Підтвердити</Button>
            </form>
        );
    }
}
