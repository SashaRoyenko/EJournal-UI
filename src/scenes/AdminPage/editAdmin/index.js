import React from 'react';
import { Wrapper, TableItem, Input, ItemTitle, Button} from './styled';

export default class Footer extends React.Component {
    render() {
        return (
            <Wrapper>
                <form>
                <table>
                    <tr>
                        <TableItem>
                            <ItemTitle>Прізвище</ItemTitle>
                            <Input type="text" placeholder="#" />
                        </TableItem>
                        <TableItem>
                            <ItemTitle>Ім'я</ItemTitle>
                            <Input type="text" placeholder="#" />
                        </TableItem>
                        <TableItem>
                            <ItemTitle>Навчальний заклад</ItemTitle>
                            <select name="#" >
                                <option value="">Навчальний заклад</option>
                            </select>
                        </TableItem>
                    </tr>
                    <tr>
                        <TableItem>
                            <ItemTitle>Електронна пошта</ItemTitle>
                            <Input type="text" placeholder="#" />
                        </TableItem>
                        <TableItem>
                            <ItemTitle>Пароль</ItemTitle>
                            <Input type="password" placeholder="#" />
                        </TableItem>
                    </tr>
                </table>
                    <Button>Підтвердити</Button>
                </form>
            </Wrapper>
        );
    }
}
