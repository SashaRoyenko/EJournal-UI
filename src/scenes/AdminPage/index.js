import React from 'react';
import Header from './Header/index';
import Aside from './Aside/index';
import Footer from './Footer/index';
import Content from './Content/index';
import { Wrapper, Main } from './styled';

export default class AdminPage extends React.Component {
  render() {
    return (
      <Wrapper>
        <Header />
        <Main>
          <Aside matchPath='/admin' />
          <Content matchPath='/admin' />
        </Main>
        <Footer />
      </Wrapper>
    );
  }
}
