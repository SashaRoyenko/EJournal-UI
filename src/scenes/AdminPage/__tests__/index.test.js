import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import AdminPage from '../index';

describe('<AdminPage />', () => {
  it('should render AdminPage', () => {
    const sut = mount(<Router><AdminPage /></Router>);

    expect(sut).toMatchSnapshot();
  });
});
