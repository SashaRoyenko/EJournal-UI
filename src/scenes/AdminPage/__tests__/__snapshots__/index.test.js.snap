// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPage /> should render AdminPage 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AdminPage>
      <styled.div>
        <div
          className="sc-fzokOt gxZieN"
        >
          <Header>
            <styled.div>
              <div
                className="sc-AxjAm krUDSd"
              >
                <Styled(Link)
                  to="/"
                >
                  <Link
                    className="sc-AxirZ hDvptI"
                    to="/"
                  >
                    <LinkAnchor
                      className="sc-AxirZ hDvptI"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="sc-AxirZ hDvptI"
                        href="/"
                        onClick={[Function]}
                      >
                        E-JOURNAL
                      </a>
                    </LinkAnchor>
                  </Link>
                </Styled(Link)>
                <styled.div>
                  <div
                    className="sc-AxiKw cnGxsf"
                  >
                    <styled.span>
                      <span
                        className="sc-AxhCb iJjSrs"
                      >
                        Ви увійшли як: 
                        <styled.span
                          highlighted={true}
                        >
                          <span
                            className="sc-AxhCb bSAOZc"
                          >
                            SuperAdmin
                          </span>
                        </styled.span>
                      </span>
                    </styled.span>
                    <styled.button>
                      <button
                        className="sc-AxhUy gLaaua"
                      >
                        Вийти
                      </button>
                    </styled.button>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </Header>
          <styled.div>
            <div
              className="sc-fzqBZW hzcYnc"
            >
              <Aside
                matchPath="/admin"
              >
                <styled.div>
                  <div
                    className="sc-AxgMl jnorjr"
                  >
                    <Styled(Link)
                      key="0"
                      to="/admin/admins"
                    >
                      <Link
                        className="sc-AxheI gHUETA"
                        to="/admin/admins"
                      >
                        <LinkAnchor
                          className="sc-AxheI gHUETA"
                          href="/admin/admins"
                          navigate={[Function]}
                        >
                          <a
                            className="sc-AxheI gHUETA"
                            href="/admin/admins"
                            onClick={[Function]}
                          >
                            Адміністратори
                          </a>
                        </LinkAnchor>
                      </Link>
                    </Styled(Link)>
                    <Styled(Link)
                      key="1"
                      to="/admin/schools"
                    >
                      <Link
                        className="sc-AxheI gHUETA"
                        to="/admin/schools"
                      >
                        <LinkAnchor
                          className="sc-AxheI gHUETA"
                          href="/admin/schools"
                          navigate={[Function]}
                        >
                          <a
                            className="sc-AxheI gHUETA"
                            href="/admin/schools"
                            onClick={[Function]}
                          >
                            Навчальні заклади
                          </a>
                        </LinkAnchor>
                      </Link>
                    </Styled(Link)>
                  </div>
                </styled.div>
              </Aside>
              <Content
                matchPath="/admin"
              >
                <styled.div>
                  <div
                    className="sc-fznKkj bXUgUD"
                  >
                    <Route
                      component={[Function]}
                      key="0"
                      path="/admin/test"
                    />
                    <Route
                      component={[Function]}
                      key="1"
                      path="/admin/add-admin"
                    />
                    <Route
                      component={[Function]}
                      key="2"
                      path="/admin/edit-admin"
                    />
                    <Route
                      component={[Function]}
                      key="0"
                      path="/admin/test"
                    />
                    <Route
                      component={[Function]}
                      key="1"
                      path="/admin/add-admin"
                    />
                    <Route
                      component={[Function]}
                      key="2"
                      path="/admin/edit-admin"
                    />
                  </div>
                </styled.div>
              </Content>
            </div>
          </styled.div>
          <Footer>
            <styled.div>
              <div
                className="sc-Axmtr kBbqSG"
              >
                <styled.span>
                  <span
                    className="sc-AxmLO hgITUO"
                  >
                    © E-JOURNAL. ROBOSH 2020
                  </span>
                </styled.span>
              </div>
            </styled.div>
          </Footer>
        </div>
      </styled.div>
    </AdminPage>
  </Router>
</BrowserRouter>
`;
